HW0/P4/P4.py

```diff
@@ -23,54 +23,148 @@
 
     # Create 3D axes for plotting
     ax = Axes3D(plt.figure())
-
     #####################
     # Part 1:
     #
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
+    s_true = np.loadtxt("P4_trajectory.txt", delimiter = ",")
+   
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    x_coords = s_true[:,0]
+    y_coords = s_true[:,1]
+    z_coords = s_true[:,2]
+   
+    ax.plot(x_coords, y_coords, z_coords,
+            '--b', label='True trajectory')
+    
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
-    #####################
+    m = np.loadtxt("P4_measurements.txt", delimiter = ",")
+    ###### print type(m)    
+    m_trans = np.transpose(m)
+
+    Cvtr = np.array([ [1/rx,0,0], [0,1/ry,0], [0,0,1/rz] ])
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    ppts = np.dot(Cvtr, m_trans)
+    ppts_trans = np.transpose(ppts)
+    #####################
+     
+    x_coords = ppts_trans[:,0]
+    y_coords = ppts_trans[:,1]
+    z_coords = ppts_trans[:,2]
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
-
+    A = np.array([ [1,0,0,dt,0,0],
+                   [0,1,0,0,dt,0],
+                   [0,0,1,0,0,dt],
+                   [0,0,0,1-c*dt,0,0],
+                   [0,0,0,0,1-c*dt,0],
+                   [0,0,0,0,0,1-c*dt] ])
```

Comment:
I defined these matrices the same way that you did.  I'll pass on a suggestion that I got on my code:
A = np.matrix(np.zeros([6,6]))
np.fill_diagonal(A, [1,1,1,1 - cdt,1 - cdt,1 - c*dt])
A[0,3] = dt
A[1,4] = dt
A[2,5] = dt